### PR TITLE
add test case for upsert

### PR DESCRIPTION
**Pull request by YiMysty**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

A few sentences describing the overall goals of the pull request.

## Related PRs or issues

List related PRs and other issues.

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java [hunk 1]
```diff
@@ -408,6 +408,9 @@ public void upsert() {
             ContentValues contentValues = generateContentValues();
             databaseManager.upsert(contentValues);
             assertEquals(1L, databaseManager.getRowCount());
+            ContentValues actual = get(databaseManager, 1);
```
Comment (YiMysty) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java]: the replace function will work when the primary key is set, explicitly use contentValues.put("oid", 1) and call the replace will work too

**Comment by nikitag-ms** [comment]
Spoke offline, Mingzi is working on some additional changes.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java [hunk 2]
```diff
@@ -209,6 +210,41 @@ public long upsert(@NonNull ContentValues values) {
...
+                if (values.containsKey(property)) {
+                    builder.appendWhere(property+ " = ?");
+                    selectionArgs.add(values.getAsString(property));
+                }
+            }
+            Cursor cursor = getCursor(builder, null, selectionArgs.toArray(new String[0]), null);
+            try {
+                ContentValues value = nextValues(cursor);
+
+                // if only contains one result, replace the value, otherwise insert directly.
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]: `/*` comments within methods.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java [hunk 3]
```diff
@@ -209,6 +210,41 @@ public long upsert(@NonNull ContentValues values) {
...
     }
 
+    /**
+     * Replaces the row, if the given property string values match the values of the row. Insert a new row if cannot find the match property values or multiple rows matches.
+     *
+     * @param values The entry to be stored.
+     * @param properties The property to be used for filter the rows.
+     * @return If a log was inserted, the database identifier. Otherwise -1.
+     */
+    public long replaceWhenStringValuesMatch(@NonNull ContentValues values, String ... properties) {
```
[the remark this answers is not in the document]
Comment (YiMysty) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]: hmmm... I did not find the file you mentioned :see_no_evil:

[the remark this answers is not in the document]
Comment (YiMysty) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]: actually changed this method to `upsert` :)

**Pull request merged**